cp: cannot stat '../../dist/traffic_ops-8.0.0-0..el8.x86_64.rpm': No such file or directory

<!--
************ STOP!! ************
If this issue identifies a security vulnerability, DO NOT submit it! Instead, contact
the Apache Traffic Control Security Team at security@trafficcontrol.apache.org and follow the
guidelines at https://apache.org/security regarding vulnerability disclosure.

- For *SUPPORT QUESTIONS*, use the #traffic-control channel on the ASF slack (https://s.apache.org/tc-slack-request)
or the Traffic Control Users mailing list (send an email to users-subscribe@trafficcontrol.apache.org to subscribe).
- Before submitting, please **SEARCH GITHUB** for a similar issue or PR
    * https://github.com/apache/trafficcontrol/issues
    * https://github.com/apache/trafficcontrol/pulls
-->

<!-- Do not submit security vulnerabilities or support requests here - see above -->
## This Bug Report affects these Traffic Control components:
<!-- delete all those that don't apply -->
- CDN in a Box

## What did you do?
<!-- If the current behavior is a bug, please provide the *STEPS TO REPRODUCE* and
include the applicable TC version.
-->
1. Download Apache Traffic Control 8.0.0 release candiate 5 <[https://dist.apache.org/repos/dist/dev/trafficcontrol/8.0.0/RC5/apache-trafficcontrol-8.0.0.tar.gz](https://dist.apache.org/repos/dist/dev/trafficcontrol/8.0.0/RC5/apache-trafficcontrol-8.0.0.tar.gz)> and ungzip it
2. Navigate to the CDN in a Box directory
3. Try to build the Traffic Ops RPM:
   ```shell
   make --debug traffic_ops/traffic_ops.rpm
   ```

## What did you expect to happen?
<!-- Describe how the bug happens -->
Traffic Ops RPM is built and placed at `infrastructure/cdn-in-a-box/traffic_ops/traffic_ops.rpm`

## What actually happened?
<!-- Describe what the behavior would be without the bug -->
The RPM builds successfully but is not copied:
```
[...]
Results in 'dist':
total 32676
-rw-r--r-- 1 user user    89774 Jan 22 16:42 build-traffic_ops.log
-rw-r--r-- 1 user user 17229352 Jan 22 16:42 traffic_ops-8.0.0-12732.0c515199.el8.src.rpm
-rw-r--r-- 1 user user 16130264 Jan 22 16:42 traffic_ops-8.0.0-12732.0c515199.el8.x86_64.rpm
cp -f "../../dist/traffic_ops-8.0.0-0..el8.x86_64.rpm" "traffic_ops/traffic_ops.rpm"
cp: cannot stat '../../dist/traffic_ops-8.0.0-0..el8.x86_64.rpm': No such file or directory
make: *** [Makefile:169: traffic_ops/traffic_ops.rpm] Error 1
```

The RPM being built is `traffic_ops-8.0.0-12732.0c515199.el8.x86_64.rpm`, but the RPM the Makefile expects is `traffic_ops-8.0.0-0..el8.x86_64.rpm` (with a missing commit hash).

The Makefile should detect that we are not in a git repository and instead get the build number from `BUILD_NUMBER`.

--- a/infrastructure/cdn-in-a-box/Makefile
+++ b/infrastructure/cdn-in-a-box/Makefile
@@ -41,17 +41,20 @@ PKG_COMMAND := $(TC_DIR)/pkg
 PKG_FLAGS := -v -$(RHEL_VERSION)
 BUILD_SUFFIX := _build
 BUILD_NUMBER := $(shell bash -c ' \
-	set -o pipefail -o xtrace; \
-	git describe \
+	set -o errexit -o pipefail -o xtrace; \
+	git rev-parse --is-inside-work-tree >/dev/null; \
+	output="$$(git describe \
 		--long \
 		--tags \
 		--match="RELEASE-[0-9].[0-9].[0-9]" \
 		--match="RELEASE-[0-9][0-9].[0-9][0-9].[0-9][0-9]" \
 		--match="v[0-9].[0-9].[0-9]" \
 		--match="v[0-9][0-9].[0-9][0-9].[0-9][0-9]" | \
-		awk -F- "{print \$$(NF-1)}"' || \
-		echo 0 \
-	).$(shell git rev-parse --short=8 HEAD)
+		awk -F- "{print \$$(NF-1)}" || \
+		echo 0)"; \
+	output+=".$$(git rev-parse --short=8 HEAD)"; \
+	echo "$$output"' || \
+	cat $(TC_DIR)/BUILD_NUMBER)
 BUILD_ARCH   := $(shell docker run --name=ciab-get-$(BASE_IMAGE)-$(RHEL_VERSION)-arch --rm $(BASE_IMAGE):$(RHEL_VERSION) rpm --eval %_arch )
 TC_VERSION := $(shell cat "$(TC_DIR)/VERSION")
 TOMCAT_VERSION := $(shell grep '^\s*TOMCAT_VERSION=' "$(TC_DIR)/traffic_router/build/build_rpm.sh"  | cut -d= -f2)

--- a/infrastructure/cdn-in-a-box/Makefile
+++ b/infrastructure/cdn-in-a-box/Makefile
@@ -40,7 +40,18 @@ TC_DIR := $(CIAB_DIR_RELATIVE)../..
 PKG_COMMAND := $(TC_DIR)/pkg
 PKG_FLAGS := -v -$(RHEL_VERSION)
 BUILD_SUFFIX := _build
-BUILD_NUMBER := $(shell bash -c 'set -o pipefail -o xtrace; git describe --long --tags --match="RELEASE-[0-9].[0-9].[0-9]" --match="RELEASE-[0-9][0-9].[0-9][0-9].[0-9][0-9]" --match="v[0-9].[0-9].[0-9]" --match="v[0-9][0-9].[0-9][0-9].[0-9][0-9]" | awk -F- "{print \$$(NF-1)}"' || echo 0).$(shell git rev-parse --short=8 HEAD)
+BUILD_NUMBER := $(shell bash -c ' \
+	set -o pipefail -o xtrace; \
+	git describe \
+		--long \
+		--tags \
+		--match="RELEASE-[0-9].[0-9].[0-9]" \
+		--match="RELEASE-[0-9][0-9].[0-9][0-9].[0-9][0-9]" \
+		--match="v[0-9].[0-9].[0-9]" \
+		--match="v[0-9][0-9].[0-9][0-9].[0-9][0-9]" | \
+		awk -F- "{print \$$(NF-1)}"' || \
+		echo 0 \
+	).$(shell git rev-parse --short=8 HEAD)
 BUILD_ARCH   := $(shell docker run --name=ciab-get-$(BASE_IMAGE)-$(RHEL_VERSION)-arch --rm $(BASE_IMAGE):$(RHEL_VERSION) rpm --eval %_arch )
 TC_VERSION := $(shell cat "$(TC_DIR)/VERSION")
 TOMCAT_VERSION := $(shell grep '^\s*TOMCAT_VERSION=' "$(TC_DIR)/traffic_router/build/build_rpm.sh"  | cut -d= -f2)

